base_local_planner/src/obstacle_cost_function.cpp

```diff
@@ -90,11 +93,15 @@ std::vector<geometry_msgs::Point> ObstacleCostFunction::getScaledFootprint(const
   if (scale != 0.0) {
     const bool fwd = traj.xv_ > 0;
     const double forward_inflation = scale * max_forward_inflation_;
-    const double sideward_inflation = scale * sideward_inflation_scale_ * max_sideward_inflation_;
     for (unsigned int i = 0; i < scaled_footprint.size(); ++i) {
       if (fwd == (scaled_footprint[i].x > 0)) {
         // assumes no sideward motion
         scaled_footprint[i].x += std::copysign(forward_inflation, scaled_footprint[i].x);
+
+        const double sideward_inflation = scaled_footprint[i].y > 0 ?
+                                              scale * sideward_inflation_scale_left_ * max_sideward_inflation_ :
+                                              scale * sideward_inflation_scale_right_ * max_sideward_inflation_;
```

Comment:
nit: could shorten to
`const double sideward_inflation = scale * max_sideward_inflation_ * (scaled_footprint[i].y > 0 ? sideward_inflation_scale_left_ : sideward_inflation_scale_right_);`